### PR TITLE
added io functions

### PR DESCRIPTION
**Pull request by kaiyingshan**

### CODE REVIEW

**Comment by nirandaperera** [comment]
@kaiyingshan I think Gregor's comment on the circular dependency is correct. We would have to take `CylonEnv` to a separate `env.py` file. Then the conda-CI should resolve IMO.

File: python/pycylon/pycylon/frame.py [hunk 1]
```diff
@@ -2307,6 +2345,157 @@ def applymap(self, func, na_action=None) -> DataFrame:
         """
         return DataFrame(self._table.applymap(func))
 
+    @staticmethod
+    def read_csv(paths: Union[str, List[str], Dict[int, Union[str, List[str]]]],
+             env: CylonEnv, **kwargs) -> DataFrame:
```
Comment (ahmet-uyar) [inline python/pycylon/pycylon/frame.py:2350]: documentation can be added.

File: python/pycylon/pycylon/frame.py [hunk 2]
```diff
@@ -2307,6 +2345,157 @@ def applymap(self, func, na_action=None) -> DataFrame:
...
+        return _read_csv_or_json(read_csv, paths, env, **kwargs)
+
+    @staticmethod
+    def read_json(paths: Union[str, List[str], Dict[int, Union[str, List[str]]]],
+             env: CylonEnv, **kwargs) -> DataFrame:
```
Comment (ahmet-uyar) [inline python/pycylon/pycylon/frame.py:2355]: documentation can be added.

File: python/pycylon/pycylon/frame.py [hunk 3]
```diff
@@ -2307,6 +2345,157 @@ def applymap(self, func, na_action=None) -> DataFrame:
...
+        def json_read_fun(path, **kwargs):
+            return DataFrame(pj.read_json(path, **kwargs))
+        return _read_csv_or_json(json_read_fun, paths, env, **kwargs)
+
+    @staticmethod
+    def read_parquet(paths: Union[str, List[str], Dict[int, Union[str, List[str]]]],
+                 env: CylonEnv,
```
Comment (ahmet-uyar) [inline python/pycylon/pycylon/frame.py:2362]: documentation can be added.

**Comment by ahmet-uyar** [comment]
python test cases can be added for all read/write functions.

**Comment by nirandaperera** [comment]
@kaiyingshan please take a pull from the master.

**Review by nirandaperera — APPROVED** [review]
LGTM! :slightly_smiling_face:

**Comment by nirandaperera** [comment]
@kaiyingshan could you please merge master here.

**Pull request merged**